nix/darwin-headerpad-max-install-names.patch

```diff
@@ -0,0 +1,20 @@
+diff --git a/build.zig b/build.zig
+index 908f9adedc..ba71894c1b 100644
+--- a/build.zig
++++ b/build.zig
+@@ -1332,6 +1332,12 @@
+             lib.linkSystemLibrary("m");
+         }
+ 
++        // Zig doesn't allocate enough space to the ELF header for
++        // `install_name_tool` to work properly.
++        if (resolved_target.result.os.tag == .macos) {
++            shared_lib.headerpad_max_install_names = true;
++        }
++
+         const static_lib = b.addStaticLibrary(.{
+             .name = "tb_client",
+             .root_source_file = b.path("src/tigerbeetle/libtb_client.zig"),
+diff --git a/darwin-headerpad-max-install-names.patch b/darwin-headerpad-max-install-names.patch
+new file mode 100644
+index 0000000000..e69de29bb2
```

Comment:
this can probably be PR'd upstream; anyone else packaging this for macOS who tries to run `install_name_tool` will run into a similar issue